### PR TITLE
JAMES-2161 Accept flags with underscore & don't throw when unsupported flag

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keywords.java [hunk 1]
```diff
@@ -38,6 +41,8 @@
 import com.google.common.collect.ImmutableSet;
 
 public class Keywords {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Keywords.class);
```
Comment (rouazana) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keywords.java]: private should be after public

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keywords.java [hunk 2]
```diff
@@ -121,12 +126,23 @@ Keywords fromOldKeyword(OldKeyword oldKeyword) {
...
+                        .map(Optional::get),
                     Stream.of(flags.getSystemFlags())
                         .map(Keyword::fromFlag))
                 .collect(Guavate.toImmutableSet()));
         }
 
+        private Optional<Keyword> asKeyword(String flagName) {
+            try {
+                return Optional.of(new Keyword(flagName));
+            } catch (Exception e) {
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keywords.java]: can't we have a finer grain exception type here ?

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keywords.java [hunk 3]
```diff
@@ -121,12 +126,23 @@ Keywords fromOldKeyword(OldKeyword oldKeyword) {
...
+                LOGGER.warn("Fail to parse {} flag", flagName);
+                return Optional.empty();
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keywords.java]: I get bored at reading this kind of code. It's time for a Try object !
what about https://github.com/poetix/ambivalence ?

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keywords.java [hunk 4]
```diff
@@ -121,12 +126,23 @@ Keywords fromOldKeyword(OldKeyword oldKeyword) {
         public Keywords fromFlags(Flags flags) {
             return fromSet(Stream.concat(
                     Stream.of(flags.getUserFlags())
-                        .map(Keyword::new),
+                        .map(this::asKeyword)
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keywords.java]: flatMap

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keywords.java (hunk 3 above)
Comment (aduprat) [reply server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keywords.java]: This one seems great too:
https://github.com/lambdista/try

We have to vote on this point, then I will update the PR.

**Pull request merged**